### DISCUSSION TITLE
"MAX_BLOCKS_PER_MULTIPROCESSOR" seems not visible from the interface

### DISCUSSION THREAD

**Discussion by Emerald01** [inducer/pycuda#346] [Q&A]
I am trying to query MAX_BLOCKS_PER_MULTIPROCESSOR from the pycuda.driver. This is an important attribute that I want to know before assigning blocks and threads. BTW, I am using Pycuda 2021.1
```
import pycuda.driver as cuda_driver

cuda_driver.Context.get_device().get_attributes()
```
But I did not see this field although the documentation has it.  And I checked pycuda backend, it indeed has this field here in `./src/wrapper/wrap_cudadrv.cpp: `, I guess written in boost
```
 .value("MAX_BLOCKS_PER_MULTIPROCESSOR", CU_DEVICE_ATTRIBUTE_MAX_BLOCKS_PER_MULTIPROCESSOR)
```
how can I query this from pycuda API then?

**Comment by inducer** [comment]
```
>>> import pycuda.autoinit
>>> import pycuda.driver as drv
>>> drv.Device(0).max_blocks_per_multiprocessor
32
```

**Reply by Emerald01** [reply]
I tried to use this way, but seems that it cannot identify this particular field
```
>>> cuda_driver.Device(0).max_blocks_per_multiprocessor

AttributeError: type object 'device_attribute' has no attribute 'MAX_BLOCKS_PER_MULTIPROCESSOR'
```
But for example, if I tried `warp_size`, it indeed works

```
>>> cuda_driver.Device(0).warp_size
32
```
The version I am using is 2021.1
```
>>>pycuda.VERSION
(2021, 1)
```

**Reply by inducer** [reply]
Likely you built PyCUDA against a header version less than CUDA 11.0. `MAX_BLOCKS_PER_MULTIPROCESSOR` was introduced in 11.0.
